src/barrier.cpp

```diff
@@ -97,7 +95,7 @@ SEXP get_rcpp_cache() {
         SEXP getNamespaceSym = Rf_install("getNamespace"); // cannot be gc()'ed  once in symbol table
         Rcpp::Shield<SEXP> RcppString(Rf_mkString("Rcpp"));
         Rcpp::Shield<SEXP> call(Rf_lang2(getNamespaceSym, RcppString));
-        Rcpp::Shield<SEXP> RCPP(Rcpp_eval(call, R_GlobalEnv));
+        Rcpp::Shield<SEXP> RCPP(Rf_eval(call, R_GlobalEnv));
```

Comment:
This should be evaluated in base or in the Rcpp namespace.

Reply:
Most (all?) of these `Rcpp_eval()` / `Rcpp_fast_eval()` actually :)

Probably best to make these changes in another PR. And merge / check that PR before this one.

Reply:
Or modify this one and test it again?

Reply:
I was thinking a different PR would make things easier should this one be reversed.

Reply:
I think you can send another PR and I will reverse this commit.

Reply:
What do you mean?  This PR is three commits.  What do you suggest reversing?

Reply:
I mean the changes in `barrier.cpp`.

Reply:
Could do.  And test that again, or first against those three?   And wasn't there a link issue?